wcfsetup/install/files/lib/system/interaction/InteractionContextMenuComponentConfiguration.class.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace wcf\system\interaction;
+
+/**
+ * Represents the configuraton for a component of an interaction content menu.
+ *
+ * @author      Marcel Werk
+ * @copyright   2001-2025 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.2
+ */
+final class InteractionContextMenuComponentConfiguration
+{
+    public function __construct(
+        public readonly string $cssClassName = '',
+        public readonly string $buttonCssClassName = '',
+        public readonly string $dropdownMenuCssClassName = '',
+        public readonly string $label = '',
+        public readonly string $tooltip = 'wcf.global.button.moreOptions',
+        public readonly string $icon = 'ellipsis',
```

Comment:
This should expect `IFontAwesomeIcon` instead, enabling both solid and brand icons.

Add a `getIconHtml()` method that just returns `$this->icon->toHtml($this->iconSize)`.

You can support a default icon by making the constructor parameter nullable and implicitly setting it to `FontAwesomeIcon::fromValue('ellipsis')`.